common/pkg/auth/authenticator.go

```diff
@@ -153,6 +165,34 @@ func (a *ExternalAuthenticator) Authenticate(r *http.Request) (string, string, e
 	return route.clusterID, route.path, nil
 }
 
+func (a *ExternalAuthenticator) authenticateService(r *http.Request, route route) error {
+	cookie, err := r.Cookie(cookieNameToken)
+	if cookie == nil || cookie.Value == "" {
+		klog.V(2).Infof("failed to get a cookie: %s", err)
+		return ErrLoginRequired
+	}
+	token := cookie.Value
+
+	v, ok := a.loginCache.get(token)
+	if ok && v == route.service {
+		return nil
```

Comment:
Wondered whether we should check if JWT is expired, but I guess we don't need as the cache expiration time is short. Could you add some comment to explain that we don't need to check the token expiration here?